### PR TITLE
Ondrej Michal-special objects update

### PR DESCRIPTION
**Pull request by OndraMichal**
Updated Intersections, Surface Results Adjustments, Surface Contacts, Result Sections, and Structure Modifications.
Test created.

unit tests: 147 passed, 9 skipped in 199.99s

### CODE REVIEW

File: UnitTests/test_SpecialObjects.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    SurfaceContactType(1, SurfaceContactPerpendicularType.FULL_FORCE_TRANSMISSION, SurfaceContactParallelType.ELASTIC_FRICTION, [2000, SurfaceContactFrictionType.FRICTION_COEFFICIENT, 0.25])
+    SurfaceContactType.FullForce(2)
+    SurfaceContactType.RigidFriction(3)
+    SurfaceContactType.ElasticFriction(4)
+    SurfaceContactType.ElasticSurface(5)
+
+    SurfaceContact(1, 1, '1', '3')
+
+    ResultSection(1,ResultSectionType.TYPE_2_POINTS_AND_VECTOR, ResultSectionResultDirection.SHOW_RESULTS_IN_GLOBAL_MINUS_X,True,[1, [1,0,0], [0,2,0], ResultSectionProjection.PROJECTION_IN_VECTOR, [1,1,1]],)
```
Comment (dogukankaratas) [inline UnitTests/test_SpecialObjects.py:99]: Result Section returns an assert.

image

**Review by dogukankaratas — CHANGES_REQUESTED** [review]
There is a bug for result section in core operations most probably.

File: UnitTests/test_SpecialObjects.py (hunk 1 above)
Comment (OndraMichal) [reply UnitTests/test_SpecialObjects.py:99]: I adjusted for XML restructuring but I see no Assert while testing with latest GM.

Comment (OndraMichal) [reply UnitTests/test_SpecialObjects.py:99]: Seems to be caused by assigned_to_surfaces and assigned_to_solids. Before both True, now True False. Assert doesn't show.

**Review by dogukankaratas — APPROVED** [review]

**Pull request merged**